docs/access_and_permissions.md

```diff
@@ -244,6 +299,61 @@ flowchart TD
     G --Pundit.policy_scope(current_user, SupportedPermission)--> H("SupportedPermissionPolicy (scope)")
 ```
 
+#### Users signin permissions create
+
+These dependencies determine whether a user can:
+
+- complete the controller action
+- grant a given user access to an app
+
+```mermaid
+flowchart TD
+    A(Users::SigninPermissionsController#create)
+    --authorize [application, user: @user], :grant_signin_permission?, policy_class: Users::ApplicationPolicy
+    --> B(Users::ApplicationPolicy#grant_signin_permission?)
+
+    A --"UserUpdate.new(@user, { supported_permission_ids: }, current_user, user_ip_address).call"
+    --> C(UserUpdate#call)
+
+    C --SupportedPermissionParameterFilter.new(current_user, user, user_params) [...] .filtered_supported_permission_ids
+    --> D(SupportedPermissionParameterFilter#filtered_supported_permission_ids)
+    D --Pundit.policy_scope(current_user, SupportedPermission)--> E("SupportedPermissionPolicy (scope)")
+```
+
+#### Users signin permissions delete
+
+These dependencies determine whether a user can:
+
+- view the confirmation screen for revoking access to a given app
+
+```mermaid
+flowchart TD
+    A(Users::SigninPermissionsController#delete)
+    --authorize [application: @application, user: @user], :remove_signin_permission?, policy_class: Users::ApplicationPolicy
+    --> B(Users::ApplicationPolicy#remove_signin_permission?)
+```
+
+#### Users signin permissions destroy
+
+These dependencies determine whether a user can:
+
+- complete the controller action
+- revoke access to a given app
+
+```mermaid
+flowchart TD
+    A(Users::SigninPermissionsController#destroy)
+    --authorize [application, user: @user], :remove_signin_permission?, policy_class: Users::ApplicationPolicy
+    --> B(Users::ApplicationPolicy#remove_signin_permission?)
+
+    A --"UserUpdate.new(@user, { supported_permission_ids: }, current_user, user_ip_address)"
```

Comment:
Looks like one of the rebases chopped off the `.call` at the end of this line, which should still be there

Reply:
oh, I did this intentionally (although inconsistently), I thought I'd made a mistake initially - we put the `UserUpdate#call` message(?) after the `UserUpdate.new` so I thought it looked a bit like this was calling it twice. Do we mean to have the `.call` chained on the end of all of these? Am obviously still a bit unclear on these diagrams 😓

Reply:
Yeah because it's the `.call` message that ends up at `UserUpdate#call`, not the `.new` message. The middle bit is always the message, the first bit is where the message is made, and the last bit is where it's handled (we repeat the implicit or explicit 'call' for everything, e.g. the `.filtered_supported_permission_ids` call below)